meson.build

```diff
@@ -15,32 +16,82 @@ if not meson.is_subproject()
     werror = true
 endif
 
+script = join_paths(meson.source_root(), 'workaround.sh')
+
+r = run_command('/bin/sh', script)
+if r.returncode() != 0
+  message('script failed')
+endif
+
+output = r.stdout().strip()
+errortxt = r.stderr().strip()
+
+prefix = get_option('prefix')
+
+################################################################################
+# Dependencies
+################################################################################
+
+if meson.version().version_compare('>=0.54.0')
+libcjson_dep     = dependency('libcjson',      version: '>=1.7.14', fallback: ['cjson'])
```

Comment:
You don't need this if/else, since most of them are doing the exact same thing.

And use of the `[provide]` section means that new enough versions of meson automatically fall back with zero configuration, so it's not necessary to add special handling in meson.build.